### ISSUE TITLE
what is formula of Box Loss and Mask Loss in YOLOv11?

### ISSUE THREAD

**Issue by carolineadicahya** [ultralytics/ultralytics#21146]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

Hello, I’m currently analyzing YOLOv11 for academic purposes. Could you please point me to the exact implementation or formula of the box loss (e.g., CIoU, SIoU) and segmentation loss (e.g., BCE, Dice) used in YOLOv11? Are there any specific changes compared to previous versions?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @carolineadicahya, thank you for your interest in Ultralytics 🚀! We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://reddit.com/r/Ultralytics) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

---

This is an automated response 🤖. An Ultralytics engineer will review your question and assist you soon!

**Comment by glenn-jocher** [comment]
Hi @carolineadicahya! For the exact formulas and implementations, you'll want to check the source code directly in the repository:

**Box Loss**: The bounding box loss implementation can be found in [`ultralytics/utils/loss.py`](https://github.com/ultralytics/ultralytics/blob/main/ultralytics/utils/loss.py) - YOLO11 primarily uses CIoU (Complete IoU) loss for box regression.

**Segmentation Loss**: For segmentation tasks, the mask loss combines BCE and Dice loss, also implemented in the same loss.py file.

The loss functions in YOLO11 are largely consistent with YOLOv8, with the main architectural improvements being in the model structure rather than the loss formulations. You can examine the `v8DetectionLoss` and `v8SegmentationLoss` classes for the complete mathematical implementations used in YOLO11.

**Comment by carolineadicahya** [comment]
ooohh i see, i think the box loss uses SmoothL1 before. thanks for the answer! i'd appreciate your response because for my thesis😁
oh i wanna ask something. does yolo11 have officiall paper like yolo12? thank youuu

**Comment by glenn-jocher** [comment]
Hi @carolineadicahya! YOLO11 doesn't have a separate research paper - it's documented through the comprehensive [Ultralytics documentation](https://docs.ultralytics.com/models/yolo11/) and the codebase itself. The model represents an evolutionary improvement over YOLOv8 with architectural refinements rather than fundamental algorithmic changes that would warrant a separate academic publication.

Just to clarify on the box loss - YOLO11 uses CIoU (Complete IoU) loss for bounding box regression, not SmoothL1. You can see this in the loss implementation where it calculates IoU-based losses that account for overlap, distance, and aspect ratio consistency. Good luck with your thesis!

**Comment by carolineadicahya** [comment]
v8DetectionLoss uses CIoU and DFL?

**Comment by glenn-jocher** [comment]
Yes, exactly! The `v8DetectionLoss` uses both CIoU and DFL (Distribution Focal Loss). CIoU handles the bounding box regression loss measuring overlap quality, while DFL is used for the box coordinate refinement to improve localization accuracy. You can see both implementations in the loss.py file where they work together for the overall detection loss calculation.
